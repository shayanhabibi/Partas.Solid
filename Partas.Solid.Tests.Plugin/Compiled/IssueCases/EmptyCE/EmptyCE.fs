module Partas.Solid.Tests.IssueCases.EmptyCE

open Partas.Solid
open Fable.Core
open Fable.Core.JsInterop

let cn (listofclasses: string list) = listofclasses |> List.fold ((+) >> id) ""
[<SolidComponent(ComponentFlag.DebugMode)>]
let Compponent (show: bool) =
    div(class' = cn [
        "testcase"
        if show then
            "showthis"
        "i'm always here"
    ]) {
        if show then "I show sometimes!"
        "I show always!"
        For(each = !![
            1;2;3
            if show then
                4
            5
        ]) {
        }
    }