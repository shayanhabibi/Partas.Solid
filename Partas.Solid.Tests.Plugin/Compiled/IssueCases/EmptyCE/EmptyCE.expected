import { fold } from "../../fable_modules/fable-library-js.5.0.0-alpha.14/List.js";
import { uncurry2 } from "../../fable_modules/fable-library-js.5.0.0-alpha.14/Util.js";
import { For } from "solid-js";

export function cn(listofclasses) {
    return fold(uncurry2((arg) => ((y) => (arg + y))), "", listofclasses);
}

export function Compponent(show) {
    return <div class={cn(["testcase", (show) ? ("showthis") : (null), "i\'m always here"])}>
        {show ? ("I show sometimes!") : (undefined)}
        I show always!
        <For each={[1, 2, 3, (show) ? (4) : (null), 5]}>
        </For>
    </div>;
}
